### DISCUSSION TITLE
How to parse a json object and retrieve it's value based on a certain condition

### DISCUSSION THREAD

**Discussion by GokulK-123** [schibsted/jslt#241] [General]
I am fairly new to JSLT. I need to parse a json object and retrieve it's value based on a certain condition. But I couldn't find a way to parse the object. Below is my code snippet - 

def func1(obj1, obj2){
let x = "a"

// Here I need to parse through every key-value pair of obj2. If the key of obj2 is equal to"a" then I need to fetch it's corresponding value and add that as a key-value pair to obj1

$obj1
}

def main(){
let output = func1(obj1,obj2)
$output
}

**Comment by larsga** [comment]
Isn't this [the same question](https://github.com/schibsted/jslt/discussions/240) you asked yesterday?

If you want the value of key `"a"` in obj2, then you can get that with `$obj2.a`.

**Reply by GokulK-123** [reply]
Nope. It's not.
I want to know as to how I can parse a json object. As I parse, I want to evaluate each key of that object to a particular condition and take appropriate action based on the condition's result. 
I am clear with the final part which is to add a property to an existing object and is what I asked yesterday.

Say for example-

let x = "India"
obj1 = {
"a":10,
"b":11
}
obj2 = {
"England": 1,
"Australia":2,
"India":3
}
// Here I want to check if any of the keys of obj2 is equal to x. If it is then get its corresponding value else ignore.

**Reply by larsga** [reply]
So `$obj2.x`, then?

**Comment by GokulK-123** [comment]
Well, I wanted to parse the whole object once and check if that key is present or not for every property in the object obj2 and then go ahead instead of directly trying to access it like `$obj2.x`. I guess what you mentioned works just the same way when that particular property is not found.

**Reply by larsga** [reply]
You get a null value if it's not found.

I don't understand why you want traverse (not parse) the entire object. Isn't that just a waste of CPU cycles?

**Comment by GokulK-123** [comment]
Yeah, that's right. Since I am new to JSLT, I didn't know what it would return if the key was not found(possibly "null" or "undefined"). To evade running into such scenario I thought I would traverse the object. Anyway, thanks for your tips.
